src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -799,6 +809,491 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_GetExternalIPv6
+            otError otErr = otDatasetGetActive(mOTInst, &activeDataset);
+            VerifyOrExit(otErr == OT_ERROR_NONE, err = MapOpenThreadError(otErr));
+            uint32_t delay = activeDataset.mDelay;
+            err            = Encode(*aWriter, TLV::ContextTag(AttributeDataElement::kCsTag_Data), delay);
+        }
+    }
+    break;
+
+    case ThreadNetworkDiagnostics::Attributes::Ids::SecurityPolicy: {
+        err = CHIP_ERROR_NOT_IMPLEMENTED;
+        // Stuct type nopt yet supported
+        // if (otDatasetIsCommissioned(mOTInst))
+        // {
+        //     otOperationalDataset activeDataset;
+        //     otError otErr = otDatasetGetActive(mOTInst, &activeDataset);
+        //     VerifyOrExit(otErr == OT_ERROR_NONE, err = MapOpenThreadError(otErr));
+        //     activeDataset.mSecurityPolicy
+        // }
+    }
+    break;
+
+    case ThreadNetworkDiagnostics::Attributes::Ids::ChannelMask: {
+        err = CHIP_ERROR_INCORRECT_STATE;
+        if (otDatasetIsCommissioned(mOTInst))
+        {
+            otOperationalDataset activeDataset;
+            otError otErr = otDatasetGetActive(mOTInst, &activeDataset);
+            VerifyOrExit(otErr == OT_ERROR_NONE, err = MapOpenThreadError(otErr));
+
+            // In the resultant Octet string, the most significant bit of the left-most byte indicates channel 0
+            // We have to bitswap the entire uint32t before converting to octet string
+            uint32_t bitSwappedChannelMask = 0;
+            for (int i = 0, j = 31; i < 32; i++, j--)
+            {
+                bitSwappedChannelMask |= ((activeDataset.mChannelMask >> j) & 1) << i;
+            }
+
+            uint8_t buffer[sizeof(uint32_t)] = { 0 };
+            Encoding::BigEndian::Put32(buffer, bitSwappedChannelMask);
+            err = Encode(*aWriter, TLV::ContextTag(AttributeDataElement::kCsTag_Data), ByteSpan(buffer));
+        }
+    }
+    break;
+
+    case ThreadNetworkDiagnostics::Attributes::Ids::OperationalDatasetComponents: {
+        // Structure not yet supported
+        // if (otDatasetIsCommissioned(mOTInst))
+        // {
+        //     *pReadLength = sizeof(otOperationalDatasetComponents);
+        //     *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(*pReadLength));
+        //     VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        //     otOperationalDataset activeDataset;
+        //     otError otErr = otDatasetGetActive(mOTInst, &activeDataset);
+        //     VerifyOrExit(otErr == OT_ERROR_NONE, err = MapOpenThreadError(otErr));
+        //     // TODO encode TLV STRUCT with content of activeDataset.mComponents
+
+        // }
+        err = CHIP_ERROR_NOT_IMPLEMENTED;
+    }
+    break;
+
+    case ThreadNetworkDiagnostics::Attributes::Ids::ActiveNetworkFaultsList: {
+        // List not yet supported
+        err = CHIP_ERROR_NOT_IMPLEMENTED;
+        break;
+    }
+
+    default: {
+        err = CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
```

Comment:
We should probably document the API entry points here in terms of the special return codes here and what they mean, so the cluster impl is not just assuming things and other platforms know what behavior is expected from them when they implement this.

Reply:
Added a brief to this method and the entry point in ThreadStackManager.h and ConnectivityManager.h